httpx/errors/errors.go

```diff
@@ -199,3 +201,45 @@ func stacktrace(err error, skip int) errors.StackTrace {
 	}
 	return stack
 }
+
+// Will recover from a panic, and throw the error away.
+//
+// Useful for when you want to test that your panic handling is working
+// correctly. If your test actually throws a panic, it just crashes right
+// there, so you can use this function to ignore the panic.
+//
+// Ex:
+// go func() {
+//   defer errors.IgnorePanic()
+//   defer errors.PushPanicToChannel(errChan)
+//
+//   DoSomethingDangerous()
+// }()
+func IgnorePanic() {
+	recover()
+}
+
+// If there's a panic, this will push the error to the given error channel.
+//
+// Useful when you want to do many things async, and want any error put in
+// a channel. A common mistake is to only handle returned errors, and forget
+// to handle panics as well, leading to errors that look like timeout errors
+// but are actually panics.
+//
+// Ex:
+// go func() {
+//   defer errors.PushPanicToChannel(errChan)
```

Comment:
So is the idea to defer this then defer the monitor?

Reply:
Yes, the defers run in reverse order, and if the previous one panics, the next one can also recover and grab the panicked error